src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -296,4 +311,71 @@ public void deleteIsland(@NotNull Island island) {
         });
     }
 
+    /**
+     * Recalculates an island value
+     *
+     * @param island The specified Island
+     */
+    public void recalculateIsland(@NotNull Island island) {
+        //Reset their value
+        IridiumSkyblock.getInstance().getBlockValues().blockValues.keySet().stream().map(material -> IridiumSkyblock.getInstance().getIslandManager().getIslandBlock(island, material)).forEach(islandBlocks -> islandBlocks.ifPresent(blocks -> blocks.setAmount(0)));
+        island.setValue(0.00);
+        //Calculate their value
+        getIslandChunks(island, IridiumSkyblockAPI.getInstance().getWorld()).thenAccept(chunks -> recalculateIsland(island, chunks.stream().map(chunk -> chunk.getChunkSnapshot(true, false, false)).collect(Collectors.toList())));
+    }
+
+    /**
+     * Recalculates the island async with specified ChunkSnapshots
+     *
+     * @param island         The specified Island
+     * @param chunkSnapshots The specified ChunkSnapshots
+     */
+    private void recalculateIsland(@NotNull Island island, @NotNull List<ChunkSnapshot> chunkSnapshots) {
+        Bukkit.getScheduler().runTaskAsynchronously(IridiumSkyblock.getInstance(), () ->
+                chunkSnapshots.forEach(chunk -> {
+                    for (int x = 0; x < 16; x++) {
+                        for (int z = 0; z < 16; z++) {
+                            if (island.isInIsland(x + (chunk.getX() * 16), z + (chunk.getZ() * 16))) {
+                                final int maxy = chunk.getHighestBlockYAt(x, z);
+                                for (int y = 0; y <= maxy; y++) {
+                                    XMaterial xMaterial = XMaterial.matchXMaterial(chunk.getBlockType(x, y, z));
+                                    if (IridiumSkyblock.getInstance().getBlockValues().blockValues.containsKey(xMaterial)) {
```

Comment:
An air check might also be a good idea as it'd be faster than Map#contains() and will probably often be used